packages/core/src/browser/tree/tree-decorator.ts

```diff
@@ -236,13 +236,29 @@ export namespace TreeDecoration {
     /**
      * Unlike caption suffixes, tail decorations appears right-aligned after the caption and the caption suffixes (is any).
      */
-    export interface TailDecoration extends CaptionAffix {
+    export interface TailDecoration {
+
+        /**
+         * The text content of the tail decoration.
+         */
+        readonly data?: string;
+
+        /**
+         * Font data for customizing the content.
+         */
+        readonly fontData?: FontData;
 
         /**
          * Optional tooltip for the tail decoration.
          */
         readonly tooltip?: string;
 
+        /**
+         * If a string, should be the name of the Font Awesome icon with out the `fa fa-` prefix, just the name, for instance `paw`.
+         * If an array, it should be the entire Font Awesome class arrray, for instance ['fa', 'fa-paw']
+         * For the existing icons, see here: https://fontawesome.com/v4.7.0/icons/.
+         */
+        readonly icon: string | string[];
```

Comment:
@thegecko shouldn't `icon` be optional?
If not I believe it will break anyone currently using the TailDecoration such as the git widget, problem widget and search in workspace widget which all use it. (CI build errors)

Reply:
Making the changes locally to set the `icon` as optional, I am able to build and see the changes after a small test :)

<div align='center'>

<img width="598" alt="Screen Shot 2019-04-04 at 9 23 28 PM" src="https://user-images.githubusercontent.com/40359487/55598281-f1922400-571f-11e9-97a7-17e0f7fef70a.png">


</div>

Reply:
> shouldn't `icon` be optional?

Something should be non-optional, right? I guess you have removed the `extends CaptionAffix` to make `data` optional.

Reply:
@vince-fugnitto You are right, I'm surprised TS was happy with that. Thanks for the catch, updated.
@kittaakos Correct, I've made data optional. I can't see a cleaner way of making this change without breaking existing implementations.

Reply:
Updated to require `data | icon | iconClass`